warehouse/utils/oidc.py

```diff
@@ -0,0 +1,27 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# For now, only RS256 is supported.
+# We probably won't need to support providers with only symmetric keys
+# (e.g. HS256) in the foreseeable future.
+VALID_ALGS = {"RS256"}
+
+# This is a map of OIDC providers supported by Warehouse.
+# The keys are human-readable names for each provider, and the values
+# are the "issuer" (i.e., `iss`) FQDN that JWTs signed by their respective
+# JWKs are checked against.
+OIDC_PROVIDERS = {
+    "github": "https://token.actions.githubusercontent.com",
+}
+
+WELL_KNOWN_OIDC_CONF = ".well-known/openid-configuration"
```

Comment:
Is this required by the spec or could it vary from one provider to the next? If it's the latter, we probably want to do something like:

```suggestion
OIDC_PROVIDERS = {
    "github": { 
        "host": "https://token.actions.githubusercontent.com",
        "conf-path": ".well-known/openid-configuration"
    }
}
```

Reply:
Yep, it's required by the spec -- every OIDC provider *must* provide that well-known URI.

(That's in contrast to the JWKS URI, which is usually under `.well-known` but isn't actually required to be.)